### DISCUSSION TITLE
Update JQuery libraries in OpenMage

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#1994] [Ideas]
Recently I started to update a frontend theme from 2016 and I was able to use the latest JQuery v3.6.0 getting only a few issues (fixed already) . 

As you may know Magento Team included JQuery library in Magento. It is located in this directory **/js/lib/jquery**. I would suggest to updated these versions by adding the latest versions 1 and 3, also the migrate scripts for both. 

This mean the actual files must be removed from distribution then added versions 1.12.4 and 3.6.0. A few template/layout files must be updated too. Any theme that uses JQuery should use this versions to avoid any security issues. A lot of themes come with their own scripts and this can create trouble.

Probably you are aware there is a strong current against JQuery and its elimination by going back to vanilla JS but I am a person who can recognize the benefits of using JQuery in the future. For almost 30 years I have been in this industry and I have seen how something is raised in glory and then demolished. We started 20 years ago with vanilla JS, then libraries were created, and then we came back. The wheel has often been reinvented and is valid in other fields.

**Comment by Danieleeffe1** [comment]
Hi Thanks for your work.

I would like to try these updates of yours on my RWD based MIMIC theme (featured in the themes section).

Can you share the steps to perform and the files to replace?

Thank you.

**Reply by addison74** [reply]
It is very simple. Find JQuery versions I mentioned before and download them into that directory. Change the old versions in RWD layout/template files and see what is happening. I recommend using migrate scripts for identifying in Console all issues. It should work without this but it is a double check, let's call it a precaution method.

**Reply by Danieleeffe1** [reply]
I thank you

I'll try

unfortunately I don't know how to use the migration scripts, at most I copy and paste following detailed tutorials, I can only edit css ;-)

**Reply by Danieleeffe1** [reply]
I am good at finding open source modules for OpenMage some I would say fundamental, but being written for Magento sometimes they need to be adjusted and now practically all of them are no longer supported

for instance:
i found this module
https://github.com/meanbee/magento-estimateddelivery/issues
very useful to insert the possibility to choose the delivery date in the checkout

could it be an idea to put it in openmage as a delivery option?

**Reply by addison74** [reply]
> unfortunately I don't know how to use the migration scripts, at most I copy and paste following detailed tutorials, I can only edit css ;-)

There are tons of resources if you search on the Internet. Here is one https://www.learningjquery.com/2020/11/what-is-jquery-migrate. And here is JQuery download page for migrate scripts. On YouTube there are videos that explain upgrading process from Jquery 1 to 3.

**Reply by addison74** [reply]
> for instance: i found this module https://github.com/meanbee/magento-estimateddelivery/issues very useful to insert the possibility to choose the delivery date in the checkout
> 
> could it be an idea to put it in openmage as a delivery option?

Sometimes in checkout process it is needed to allow customers to write instructions or picking up delivery dates. These options can be added to OpenMage as long as they can be enabled/disabled. The problem I had many times when I wanted to add features to the code was not to load it unnecessarily because not everyone needs them.

**Reply by Danieleeffe1** [reply]
I thank you for your support

**Comment by addison74** [comment]
This was achieved in https://github.com/OpenMage/magento-lts/pull/3204.
